source/features/resolve-conflict.tsx

```diff
@@ -0,0 +1,186 @@
+			const codeMirrorElement: EditorDOMElement = document.querySelector('.CodeMirror');
+			this.editor = codeMirrorElement.CodeMirror;
+
+			/**
+             * Listen to swapDoc event in order to find out file change
+             */
+			this.editor.on('swapDoc', () => {
+				// Add 200 milisecond delay, we need keep calm until codemirror updates its content
+				setTimeout(() => {
+					const conflicts = [...document.querySelectorAll('.CodeMirror .rg-conflict')];
+					if (conflicts.length === 0) {
+						this.processConflicts();
+					}
+				}, 200);
+			});
+		}
+
+		/**
+         * Process editor lines and find conflict lines, assign class to each of lines
+         * and add resolver widget to the first line of conflict
+         */
+		processConflicts() {
+			const lines = [...document.querySelectorAll('.CodeMirror .conflict-gutter-marker')];
+			let stepBranch = '';
+			lines.forEach(line => {
+				const lineNumber = parseInt(line.parentElement.parentElement.innerText, 10) - 1;
+				let cl = `rg-conflict c-${this.conflictNumber} `;
+				if (line.classList.contains('js-start')) {
+					cl += 'rg-separator rg-start incoming-branch current-branch';
+					stepBranch = 'incoming-branch';
+					this.addWidgetToLine(this.conflictNumber, lineNumber);
+				}
+
+				if (line.classList.contains('js-middle')) {
+					cl += 'rg-separator rg-middle incoming-branch current-branch';
+					stepBranch = 'current-branch';
+				}
+
+				if (line.classList.contains('js-end')) {
+					cl += 'rg-separator rg-end incoming-branch current-branch';
+					stepBranch = '';
+					this.conflictNumber += 1;
+				}
+
+				if (line.classList.contains('js-line')) {
+					cl += 'rg-middle ' + stepBranch;
+				}
+
+				this.editor.addLineClass(lineNumber, '', cl);
+			});
+		}
+
+		/**
+         * Create and add widget to a line
+         * @param conflictNumber
+         * @param lineNumber
+         */
+		addWidgetToLine(conflictNumber: number, lineNumber: number) {
+			const widget = this.newWidget(conflictNumber);
+			this.editor.addLineWidget(lineNumber, widget, {above: true});
+		}
+
+		/**
+         * Create and return conflict resolve widget for specific conflict
+         * @param conflictNumber
+         */
+		newWidget(conflictNumber: number) {
+			const widget = document.createElement('div');
+			widget.classList.add('rg-resolver');
+			widget.style.fontWeight = 'bold';
+
+			const link = (branch, conflictNumber) => {
+				const link = document.createElement('a');
+				link.href = `#accept${branch}`;
+				link.textContent = `Accept ${branch} Change`;
+				link.addEventListener('click', e => {
+					e.preventDefault();
+					this.acceptBranch(branch, conflictNumber);
+				});
+				return link;
+			};
+
+			widget.append(link('Current', conflictNumber));
+			widget.append(document.createTextNode(' | '));
+			widget.append(link('Incoming', conflictNumber));
+			widget.append(document.createTextNode(' | '));
+			widget.append(link('Both', conflictNumber));
```

Comment:
These can be merged in a single `.append` call. It accepts multiple parameters and even plain strings (they’ll be automatically conveyed to text nodes)